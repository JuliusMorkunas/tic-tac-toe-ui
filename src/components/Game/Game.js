import React from 'react';
import s from './Game.module.scss';
import Message from '../Message';
import Reset from '../Reset';
import Board from '../Board';
import ActionLog from '../ActionLog';

const Game = ({ isLoaded, board, turn, winner, draw, actions = [], playerAction, resetGame }) => {
  const handlePlayerAction = ({ row, cell }) => {
    !board[row][cell] && !winner && playerAction({ player: turn, row, cell });
  };

  const handleGameReset = e => {
    e.preventDefault();
    resetGame();
  };

  return (
    <div className={s.Game}>
      {winner && (
        <>
          <Message message="Winner winner, chicken dinner!" />
          <Reset message="Again!" onClick={handleGameReset} />
        </>
      )}
      {draw && (
        <>
          <Message message="It's a draw. :(" />
          <Reset message="Try again..." onClick={handleGameReset} />
        </>
      )}
      {!draw && !winner && (
        <>
          <Message message={`Turn: ${turn}`} />
          <Reset message="Restart." disabled={!actions.length} onClick={handleGameReset} />
        </>
      )}
      <Board board={board} winner={winner} handlePlayerAction={handlePlayerAction} />
      <ActionLog actions={actions} />
    </div>
  );
};

export default Game;
