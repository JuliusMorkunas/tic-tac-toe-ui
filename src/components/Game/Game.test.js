import React from 'react';
import renderer from 'react-test-renderer';
import Game from './Game';

it('Game renders correctly', () => {
  const tree = renderer.create(<Game />).toJSON();
  expect(tree).toMatchSnapshot();
});
