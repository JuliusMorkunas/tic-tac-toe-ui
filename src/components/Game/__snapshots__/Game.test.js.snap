// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders correctly 1`] = `
<div
  className="Game"
>
  <p
    className="Message"
  >
    Turn: undefined
  </p>
  <div
    className="Restart IsDisabled"
    onClick={[Function]}
    role="button"
  >
    Restart.
  </div>
  <div
    className="Board"
  />
</div>
`;
